File: Split_a_String_in_Balanced_Strings.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        balance = 0
        no_of_substrings = 0
        for i in s:
            if(i =="L"):
                balance = balance +1
            else:
                balance=  balance -1
            if(balance ==0):
                no_of_substrings = no_of_substrings+1
        return no_of_substrings

File: Split_a_String_in_Balanced_Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int no_of_substrings = 0;
        int balance = 0;
        for (int i =0;i < s.size();i++){
            if(s[i]=='L'){
                balance++;
            }
            else{
                balance--;
            }
            if(balance ==0){
                no_of_substrings++;
            }
        }
        return no_of_substrings;
        
    }
};
